diff --git a/install.sh b/install.sh
index 7b47f93..bba8b0c 100755
--- a/install.sh
+++ b/install.sh
@@ -10,15 +10,15 @@ cd $DIR
 GIT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
 GIT_COMMIT_HASH="$(git log -1 --format=%h)"
 
-if pyang -v | grep -q '1.7.3'; then
-  	echo "Pyang version 1.7.3 is already installed"
-  	sudo cp $(pwd)/pyang/plugins/swagger.py /usr/local/lib/python2.7/dist-packages/pyang-1.7.3-py2.7.egg/pyang/plugins/swagger.py
+if pyang -v | grep -q '1.7.8'; then
+  	echo "Pyang version 1.7.8 is already installed"
+  	sudo cp $(pwd)/pyang/plugins/swagger.py /usr/local/lib/python2.7/dist-packages/pyang-1.7.8-py2.7.egg/pyang/plugins/swagger.py
 else
 	sudo pip install -r requirements.txt
 	sudo rm -rf /tmp/pyang-swagger
 	mkdir /tmp/pyang-swagger
 
-	wget https://github.com/mbj4668/pyang/archive/pyang-1.7.3.tar.gz -O /tmp/pyang-swagger/pyang.tar.gz
+	wget https://github.com/mbj4668/pyang/archive/pyang-1.7.8.tar.gz -O /tmp/pyang-swagger/pyang.tar.gz
 	mkdir /tmp/pyang-swagger/pyang
 	tar -zxf /tmp/pyang-swagger/pyang.tar.gz -C /tmp/pyang-swagger/pyang/ --strip-components=1
 
